weed/filer/ydb/ydb_store_kv.go

```diff
@@ -0,0 +1,79 @@
+//go:build ydb
+// +build ydb
+
+package ydb
+
+import (
+	"context"
+	"fmt"
+	"github.com/ydb-platform/ydb-go-sdk/v3/table/options"
+
+	"github.com/chrislusf/seaweedfs/weed/filer"
+	"github.com/chrislusf/seaweedfs/weed/filer/abstract_sql"
+	"github.com/chrislusf/seaweedfs/weed/util"
+	"github.com/ydb-platform/ydb-go-sdk/v3/table"
+	"github.com/ydb-platform/ydb-go-sdk/v3/table/result/named"
+	"github.com/ydb-platform/ydb-go-sdk/v3/table/types"
+)
+
+func (store *YdbStore) KvPut(ctx context.Context, key []byte, value []byte) (err error) {
+	dirStr, dirHash, name := abstract_sql.GenDirAndName(key)
+	fileMeta := FileMeta{dirHash, name, dirStr, value}
+	return store.DB.Table().Do(ctx, func(ctx context.Context, s table.Session) (err error) {
+		_, _, err = s.Execute(ctx, rwTX, *withPragma(&store.tablePathPrefix, insertQuery),
+			fileMeta.queryParameters(0),
+			options.WithQueryCachePolicy(options.WithQueryCachePolicyKeepInCache()))
+		if err != nil {
+			return fmt.Errorf("kv put execute %s: %v", util.NewFullPath(dirStr, name).Name(), err)
+		}
+		return nil
+	})
+}
+
+func (store *YdbStore) KvGet(ctx context.Context, key []byte) (value []byte, err error) {
+	dirStr, dirHash, name := abstract_sql.GenDirAndName(key)
+	valueFound := false
+	err = store.DB.Table().Do(ctx, func(ctx context.Context, s table.Session) error {
+		_, res, err := s.Execute(ctx, roTX, *withPragma(&store.tablePathPrefix, findQuery),
+			table.NewQueryParameters(
+				table.ValueParam("$dir_hash", types.Int64Value(dirHash)),
+				table.ValueParam("$name", types.UTF8Value(name))),
+			options.WithQueryCachePolicy(options.WithQueryCachePolicyKeepInCache()))
+		if err != nil {
+			return fmt.Errorf("kv get execute %s: %v", util.NewFullPath(dirStr, name).Name(), err)
+		}
+		defer func() { _ = res.Close() }()
+		for res.NextResultSet(ctx) {
+			for res.NextRow() {
+				if err := res.ScanNamed(named.OptionalWithDefault("meta", &value)); err != nil {
+					return fmt.Errorf("scanNamed %s : %v", util.NewFullPath(dirStr, name).Name(), err)
+				}
+				valueFound = true
+				return nil
+			}
+		}
+		return res.Err()
+	})
+
+	if !valueFound {
+		return nil, filer.ErrKvNotFound
+	}
+
+	return value, nil
+}
+
+func (store *YdbStore) KvDelete(ctx context.Context, key []byte) (err error) {
+	dirStr, dirHash, name := abstract_sql.GenDirAndName(key)
+	return store.DB.Table().Do(ctx, func(ctx context.Context, s table.Session) (err error) {
+		_, _, err = s.Execute(ctx, rwTX, *withPragma(&store.tablePathPrefix, insertQuery),
```

Comment:
insertQuery has 4 non optional parameters, you send two only.
Query will fail if not send values for non optional parameters.